bokeh/io.py

```diff
@@ -413,6 +414,8 @@ def _get_save_args(state, filename, resources, title):
     if filename is None:
         warn = False
         filename = _detect_filename("html")
+        if not os.access(filename, os.W_OK):
```

Comment:
`os.access("non_existent_filename", os.W_OK)` returns `False`, so you have to proceed with the solution proposed in the issue, i.e. check `W_OK | X_OK` for the base directory.